core/workflow/rolebase.go

```diff
@@ -350,4 +350,12 @@ func (r *roleBase) CollectInboundChannels() (channels []channel.Inbound) {
 		channels = channel.MergeInbound(r.Bind, r.parent.CollectInboundChannels())
 	}	
 	return
+}
+
+func (r *roleBase) GetTaskPID() string {
```

Comment:
Related to the other comments, perhaps `GetTaskPID` could only be a member of `taskRole`, or alternatively, it could be outside the role system entirely and come directly from `task.Task`.